### PR TITLE
http3: deprecate envoy.reloadable_features.conn_pool_new_stream_with_early_data_and_http3

### LINKED ISSUE #25011: envoy_reloadable_features_enable_compression_bomb_protection deprecation
Your change #733 (retry: retry immediately on 425 response code and H3 stream reset) introduced a runtime guarded feature. It has been 6 months since the new code has been exercised by default, so it's time to remove the old code path. This issue tracks source code cleanup so we don't forget.

### PR DESCRIPTION
**Pull request by danzh2010**
Signed-off-by: Dan Zhang <danzh@google.com>

Commit Message: deprecate above runtime guard

Risk Level: low, no behavior change
Testing: existing tests pass
Docs Changes: changelog
Release Notes: N/A
Platform Specific Features: N/A
Fixes #25011
Deprecated: ``envoy.reloadable_features.conn_pool_new_stream_with_early_data_and_http3``

### CODE REVIEW

File: changelogs/current.yaml [hunk 1]
```diff
@@ -10,7 +10,9 @@ bug_fixes:
 # *Changes expected to improve the state of the world and are unlikely to have negative effects*
 
 removed_config_or_runtime:
-# *Normally occurs at the end of the* :ref:`deprecation period <deprecated>`
```
Comment (RyanTheOptimist) [inline changelogs/current.yaml:13]: nit: should we restore this line?

**Review by RyanTheOptimist — COMMENTED** [review]
The change looks great. One question, though. How did you know that this was the right time to delete this runtime guard?

**Comment by danzh2010** [comment]
> How did you know that this was the right time to delete this runtime guard?

It is Envoy policy to deprecate runtime guard in ~6 months. At least during the past 6 months, the new code (whether exercised or not) didn't cause production issues. 

I would **think** future usage would use the config knob to turn on/off HTTP/3 early data support via this extension: https://www.envoyproxy.io/docs/envoy/latest/api-v3/extensions/early_data/v3/default_early_data_policy.proto

**Comment by RyanTheOptimist** [comment]
> > How did you know that this was the right time to delete this runtime guard?
> 
> It is Envoy policy to deprecate runtime guard in ~6 months. At least during the past 6 months, the new code (whether exercised or not) didn't cause production issues.
> 
> I would **think** future usage would use the config knob to turn off HTTP/3 early data support via this extension: https://www.envoyproxy.io/docs/envoy/latest/api-v3/extensions/early_data/v3/default_early_data_policy.proto if anything goes wrong.

Ah, It's been 6 months already. LGTM

**Review by RyanTheOptimist — DISMISSED** [review]

File: changelogs/current.yaml (hunk 1 above)
Comment (ggreenway) [reply changelogs/current.yaml:13]: Yes, please leave the comment line

**Review by ggreenway — CHANGES_REQUESTED** [review]
LGTM, aside from comment in changelog

File: changelogs/current.yaml (hunk 1 above)
Comment (danzh2010) [reply changelogs/current.yaml]: done

**Review by RyanTheOptimist — APPROVED** [review]

**Comment by RyanTheOptimist** [comment]
LGTM. @ggreenway, look good to you?

**Review by ggreenway — APPROVED** [review]

**Pull request merged**